dask/blockwise.py

```diff
@@ -917,21 +919,12 @@ def make_blockwise_graph(
         if deserializing:
             deps.update(func_future_args)
             args += list(func_future_args)
-            if kwargs:
-                val = {
-                    "function": dumps_function(apply),
-                    "args": warn_dumps(args),
-                    "kwargs": warn_dumps(kwargs2),
-                }
-            else:
-                val = {"function": func, "args": warn_dumps(args)}
+        if kwargs:
+            val = (apply, func, args, kwargs2)
         else:
-            if kwargs:
-                val = (apply, func, args, kwargs2)
-            else:
-                args.insert(0, func)
-                val = tuple(args)
-        dsk[out_key] = val
+            args.insert(0, func)
+            val = tuple(args)
+        dsk[out_key] = to_serialize(val) if deserializing else val
```

Comment:
Note that, if [distributed#](https://github.com/dask/distributed/pull/4641) gets in, we can do something like:

```
dsk[out_key] = Serialized(*serialize(val, iterate_collection=True)) if deserializing else val
```

and then the `SerializedFunction` wrapper becomes completely unnecessary :)

Reply:
Just a heads up, when using https://github.com/dask/distributed/pull/4575, `to_serialize` should work here:

```python
dsk[out_key] = to_serialize(val) if deserializing else val
```

Reply:
Ah, cool - thanks!  I'll give that a try.

Reply:
That PR still has other issues though. So something to be aware of

Reply:
>That PR still has other issues though. So something to be aware of

Thanks for the heads up John.  Is there anything specific that we can do to help?  It seems that Blockwise-IO serialization may be blocked in dask/dask until something like your PR gets through in dask/distributed :)